components/service-operator/apis/database/v1beta1/postgres_types.go

```diff
@@ -16,60 +16,280 @@ limitations under the License.
+	}
+
+	for i := 0; i < InstanceCount; i++ {
+		template.Resources[fmt.Sprintf("%s%d", PostgresResourceInstance, i)] = &cloudformation.AWSRDSDBInstance{
+			DBClusterIdentifier:  cloudformation.Ref(PostgresResourceCluster),
+			DBInstanceClass:      coalesce(p.Spec.AWS.InstanceType, DefaultClass),
+			Engine:               Engine,
+			PubliclyAccessible:   false,
+			DBParameterGroupName: cloudformation.Ref(PostgresResourceParameterGroup),
+			Tags:                 tags,
+			DBSubnetGroupName:    cloudformation.Ref(DBSubnetGroupNameParameterName),
+		}
+	}
+
+	template.Resources[PostgresResourceClusterParameterGroup] = &cloudformation.AWSRDSDBClusterParameterGroup{
+		Description: "GSP Service Operator Cluster Parameter Group",
+		Family:      Family,
+		Parameters: map[string]string{
+			"timezone": "UTC",
+		},
+		Tags: tags,
+	}
+
+	template.Resources[PostgresResourceParameterGroup] = &cloudformation.AWSRDSDBParameterGroup{
+		Description: "GSP Service Operator Parameter Group",
+		Family:      Family,
+		Parameters: map[string]string{
+			"application_name": p.GetStackName(),
+		},
+		Tags: tags,
+	}
+
+	template.Resources[PostgresResourceIAMPolicy] = &cloudformation.AWSIAMPolicy{
+		PolicyName: cloudformation.Join("-", []string{"postgres", "access", cloudformation.Ref(PostgresResourceCluster)}),
+		PolicyDocument: cloudformation.NewRolePolicyDocument(
+			[]string{
+				cloudformation.Join(":", []string{
+					"arn",
+					"aws",
+					"rds",
+					cloudformation.Ref("AWS::Region"),
+					cloudformation.Ref("AWS::AccountId"),
+					"cluster",
+					cloudformation.Ref(PostgresResourceCluster),
+				}),
+			},
+			[]string{"rds-data:*"},
+		),
+		Roles: []string{
+			cloudformation.Ref(IAMRoleNameParameterName),
+		},
+	}
+
+	template.Outputs[PostgresEndpoint] = map[string]interface{}{
+		"Description": "Postgres Endpoint used by the application to perform connection.",
+		"Value":       cloudformation.GetAtt(PostgresResourceCluster, "Endpoint.Address"),
+	}
+
+	template.Outputs[PostgresReadEndpoint] = map[string]interface{}{
+		"Description": "Postgres reader Endpoint used by the application to perform connection.",
+		"Value":       cloudformation.GetAtt(PostgresResourceCluster, "ReadEndpoint.Address"),
+	}
+
+	template.Outputs[PostgresPort] = map[string]interface{}{
+		"Description": "Postgres Port used by the application to perform connection.",
+		"Value":       cloudformation.GetAtt(PostgresResourceCluster, "Endpoint.Port"),
+	}
 
-	Spec   PostgresSpec   `json:"spec,omitempty"`
-	Status PostgresStatus `json:"status,omitempty"`
+	template.Outputs[PostgresUsernameOutputName] = map[string]interface{}{
+		"Description": "Postgres master username",
+		"Value":       cloudformation.Ref(PostgresUsername),
+	}
+
+	template.Outputs[PostgresPasswordOutputName] = map[string]interface{}{
+		"Description": "Postgres master password",
+		"Value":       cloudformation.Ref(PostgresPassword),
```

Comment:
We can't output this from a CloudFormation stack. The postgres handling in the service operator should only get this (and possibly the username) from the existing k8s Secret, to update into the new secret when we have outputs like endpoint and port.

Reply:
Excellent point, thanks @Krenair  ...

I' think I can rejig the logic so that `Secrets` are not populated from cloudformation Outputs but by requesting a list of Secret Manager keys.

This would have the following benefits:

* Can still be implemented as a generic interface
* Will mean that secret generation can be removed from the operator entirely
* Will mean we could take advantage of rotation features of cloudformation/secret-manager in future

... I'll have a go at this now.

Reply:
Well we still need to generate the values when we create the stack, but I'm sure this can be done with a generic interface too.

Reply:
I addressed this by using secretsmanager :+1: